eclipse-collections/src/main/java/org/eclipse/collections/impl/collector/Collectors2.java

```diff
@@ -423,6 +761,15 @@ private Collectors2()
                 EMPTY_CHARACTERISTICS);
     }
 
+    /**
+     * <p>Returns the elements as an MutableMultimap grouping each element using the specified groupBy Function and
+     * converting each element to the value returned by applying the specified Function valueFunction.</p>
+     * <p>Examples:</p>
+     * {@code MutableMultimap<String, String> multimap1 =
+     * Interval.oneTo(5).stream().collect(Collectors2.groupBy(Object::toString, Object::toString, Multimaps.mutable.list::empty));}<br>
+     * {@code MutableMultimap<String, String> multimap2 =
+     * Interval.oneTo(5).reduceInPlace(Collectors2.groupBy(Object::toString, Object::toString, Multimaps.mutable.list::empty));}
+     */
     private static <T, K, V, R extends MutableMultimap<K, V>> Collector<T, ?, R> groupBy(
```

Comment:
This is not regarding Javadoc and it can be a separate change. 

The naming of this method `groupBy` is different to what we propose in Issue #47 `groupByAndCollect`. I think the naming suggestion in #47 makes sense and we can change this before releasing 8.0.  

Same case for `groupByImmutable` method.